service/history/task/transfer_active_task_executor.go

```diff
@@ -1015,7 +1015,7 @@ func (t *transferActiveTaskExecutor) requestCancelExternalExecutionCompleted(
 	err := t.updateWorkflowExecution(ctx, wfContext, true,
 		func(ctx context.Context, mutableState execution.MutableState) error {
 			if !mutableState.IsWorkflowExecutionRunning() {
-				return &types.EntityNotExistsError{Message: "Workflow execution already completed."}
+				return &types.WorkflowExecutionAlreadyCompletedError{Message: "Workflow execution already completed."}
```

Comment:
Not related to change on this line.  Just want to make sure that: is the error handling logic able to handle the new error type returned from `historyClient` calls in this file?

Reply:
> is the error handling logic able to handle the new error type 

Could you explain this a bit more? How can I check this?

Reply:
Probably find all calls to history client in this file and see if the call might return already completed error or not. If so, maybe check if it is correctly treated as non-retryable and the same way as the entity not exists error returned previously.

Reply:
As far as this file goes, there was only one place to be handled in the caller to `applyParentClosePolicy`. I fixed this one.

We return this error also from `requestCancelExternalExecutionWithRetry` and `signalExternalExecutionWithRetry` which forward the error to their external callers from `Execute`. And those seem to be handled in task.go:HandleErr which I also changed (you'll see in the updated PR).

I couldn't find other places but let me know if there are other places if you can think of other places.

Reply:
Probably also update the logic in `createFirstDecisionTask`? It's ok if it's not updated I think as the logic in `task.go` will handle it, but to avoid confusion, probably also update that?